qlib/model/riskmodel/structured.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+from typing import Union
+from sklearn.decomposition import PCA, FactorAnalysis
+
+from qlib.model.riskmodel import RiskModel
+
+
+class StructuredCovEstimator(RiskModel):
+    """Structured Covariance Estimator
+
+    This estimator assumes observations can be predicted by multiple factors
+        X = FB + U
+    where `F` can be specified by explicit risk factors or latent factors.
+
+    Therefore the structured covariance can be estimated by
+        cov(X) = F cov(B) F.T + cov(U)
+
+    We use latent factor models to estimate the structured covariance.
+    Specifically, the following latent factor models are supported:
+        - `pca`: Principal Component Analysis
+        - `fa`: Factor Analysis
+
+    Reference: [1] Fan, J., Liao, Y., & Liu, H. (2016). An overview of the estimation of large covariance and
+    precision matrices. Econometrics Journal, 19(1), C1–C32. https://doi.org/10.1111/ectj.12061
+    """
+
+    FACTOR_MODEL_PCA = "pca"
+    FACTOR_MODEL_FA = "fa"
+    NAN_OPTION = "fill"
+
+    def __init__(
+            self,
+            factor_model: str = "pca",
+            num_factors: int = 10,
+            assume_centered: bool = False,
+            scale_return: bool = True,
+    ):
+        """
+        Args:
+            factor_model (str): the latent factor models used to estimate the structured covariance (`pca`/`fa`).
+            num_factors (int): number of components to keep.
+            assume_centered (bool): whether the data is assumed to be centered.
+            scale_return (bool): whether scale returns as percentage.
+        """
+        super().__init__(self.NAN_OPTION, assume_centered, scale_return)
```

Comment:
`nan_option` should be passed in `StructuredCovEstimator.__init__` to match its parent's interface. If only `nan_option=fill`  is supported in this model, you should add argument validation like `assert` to notify user.

Reply:
Plz put `nan_option`, assume_centered`, and `scale_return` to `**kwargs` and call `super().__init__(**kwargs)` to init them. such that we don't need to chang codes here again if we plan to modify our base model's interface in the future.

Reply:
Resolved in commit `4d5a30b`.